wire-runtime/src/test/java/com/squareup/wire/protos/redacted/NotRedacted.java

```diff
@@ -50,13 +90,22 @@ public boolean equals(Object other) {
   public int hashCode() {
     int result = hashCode;
     if (result == 0) {
-      result = a != null ? a.hashCode() : 0;
+      result = result * 37 + (a != null ? a.hashCode() : 0);
       result = result * 37 + (b != null ? b.hashCode() : 0);
       hashCode = result;
     }
     return result;
   }
 
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder();
+    toStringValue(sb, "a", a);
+    toStringValue(sb, "b", b);
+    sb.insert(0, "NotRedacted{");
```

Comment:
`toStringValue` checks against 0 to determine whether to comma-separate. Without this it has no way of knowing whether to add the comma without passing in the length of this initial string.